manifests/init.pp

```diff
@@ -837,6 +838,9 @@
   validate_re($sshd_pam_access, $valid_pam_access_values,
     "pam::sshd_pam_access is <${sshd_pam_access}> and must be either 'required', 'requisite', 'sufficient', 'optional' or 'absent'.")
 
+  $limits_fragments_hiera_merge_real = str2bool($limits_fragments_hiera_merge)
```

Comment:
I think you need a conditional around this in case the variable is already a boolean.

something like

```puppet
if is_string($limits_fragments_hiera_merge) == true {
  $limits_fragments_hiera_merge_real = str2bool($limits_fragments_hiera_merge)
} else {
  $limits_fragments_hiera_merge_real = $limits_fragments_hiera_merge
}
validate_bool($limits_fragments_hiera_merge_real)
```